wip queuefeed skel

asg0451 opened the pull request.

asg0451 committed:
wip

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/157081)
<!-- Reviewable:end -->

asg0451 committed:
fix up and move stuff around

asg0451 committed:
add oid entry

asg0451 committed:
make qm

asg0451 committed:
working

asg0451 committed:
rangefeed

jeffswenson committed:
queuefeed: implement simple assignment algorithm

This implements a simple assignment algorithm where there is one
partition that is assigned on a first come first serve basis.

jeffswenson committed:
queuefeed: add wrapper for partitions table

aerfrei committed:
read from rangefeed

aerfrei committed:
make sure it doesn't panic

aerfrei committed:
queuefeed: have crdb_internal.select_from_queue_feed return rows

Instead of returning a single array as json for the events stored
in our queue, now return a row for each row so that we get

root@127.0.0.1:26257/demoapp/movr> SELECT crdb_internal.select_from_queue_feed('foo', 3);
  crdb_internal.select_from_queue_feed
----------------------------------------
  {"f1": "1", "f2": "2", "f3": "3"}
  {"f1": "4", "f2": "5", "f3": "6"}
  {"f1": "7", "f2": "8", "f3": "9"}

instead of

root@127.0.0.1:26257/demoapp/movr> SELECT crdb_internal.select_from_queue_feed('foo', 3);
                                                             crdb_internal.select_from_queue_feed
--------------------------------------------------------------------------------------------------------------------------------------------------------------
  {"{\"f1\": \"1\", \"f2\": \"2\", \"f3\": \"3\"}","{\"f1\": \"11\", \"f2\": \"22\", \"f3\": \"33\"}","{\"f1\": \"111\", \"f2\": \"222\", \"f3\": \"333\"}"}
(1 row)

aerfrei committed:
Merge pull request #3 from aerfrei/qf-table-based

Qf table based

aerfrei committed:
add a table to track existing queues and add test

aerfrei committed:
Merge pull request #5 from aerfrei/qf-feeds-table

add a table to track existing queues and add test

asg0451 committed:
Merge pull request #6 from asg0451/wip-queuefeed-rangefeed-and-stuff

asg0451 committed:
fix build

asg0451 committed:
restore functionality

asg0451 committed:
fix test

asg0451 committed:
cleanup

asg0451 committed:
wip

asg0451 committed:
use session context

asg0451 committed:
fix

asg0451 committed:
add test

asg0451 committed:
clean

asg0451 committed:
store readers

asg0451 committed:
Merge pull request #7 from asg0451/fix-build

fix build and restore basic functionality and fix test

asg0451 committed:
cleanup

asg0451 committed:
Merge pull request #8 from asg0451/fix-ctx-and-other-stuff

fix context usage (restore cancellability)

asg0451 committed:
Merge pull request #9 from asg0451/store-readers

store readers

asg0451 committed:
rollback support

asg0451 committed:
Merge pull request #11 from asg0451/wip-queuefeed-rollback

rollback support

jeffswenson committed:
queuefeed: create partition cache

This change includes the partition cache. The idea is this will be
refreshed based on the rangefeed and local transactions that update the
table. The core assignment algorithms were implemented against the cache
which allows us to quickly test a large number of random scenarios.

The production assigner will basically mimic the simulation assigner,
but it will use txns to update rows in the db before applying the
updates to the cache.

aerfrei committed:
queuefeed: Add checkpointing logic

Epic: none

jeffswenson committed:
queuefeed: add a smoke test

asg0451 committed:
fix context bug

asg0451 committed:
Merge pull request #13 from asg0451/fix-context-more

fix context bug

jeffswenson committed:
queuefeed: wire up the assigner

asg0451 committed:
rudimentary reassignments

asg0451 committed:
Merge pull request #17 from asg0451/refresh-assignment

rudimentary refresh assignment

asg0451 committed:
Create a partition per range

asg0451 committed:
move reader management to conn executor

asg0451 committed:
cleanup

aerfrei committed:
Merge pull request #14 from aerfrei/qf-cursor

queuefeed checkpointing

asg0451 committed:
Merge pull request #20 from asg0451/move-connex

Move reader mgmt to conn executor

asg0451 committed:
fix test

jeffswenson committed:
queuefeed: add smoke test with multiple partitions

Add a test for multiple ranges. Rewrite the span splitting logic so that
it follows the pattern established by backup.

asg0451 committed:
watch all my partitions

asg0451 committed:
Merge pull request #23 from asg0451/watch-all-parts

watch all my partitions

asg0451 committed:
unassign dead sessions

jeffswenson committed:
queuefeed: wire up partition cache to the assigner

asg0451 committed:
Merge pull request #25 from asg0451/reset-unalive-assignments

unassign dead sessions

jeffswenson committed:
queuefeed: create a multi-reader smoke test

jeffswenson committed:
queuefeed: call close on manager in tests

aerfrei committed:
queuefeed: support partitions in checkpointing

Epic: none

aerfrei committed:
queuefeed: use frontier for tracking partition progress

Epic: none

aerfrei committed:
queuefeed: set the cursor to the queue creation time on creation

Epic: none

aerfrei committed:
Merge pull request #24 from aerfrei/qf-partition-checkpointing-support

queuefeed: support checkpoints per partition

aerfrei committed:
add builtin for starting a changefeed from a cursor

aerfrei committed:
queuefeed: create queuefeed from table name

Epic: none

aerfrei committed:
Merge pull request #28 from aerfrei/qf-cursor-builtin

add builtin for starting a changefeed from a cursor

jeffswenson committed:
roachtest: add queuefeed roachtest

aerfrei committed:
Merge pull request #29 from aerfrei/qf-builtin-to-take-table-name

create queuefeed from name
